### DISCUSSION TITLE
Full text search on multiple tables?

### DISCUSSION THREAD

**Discussion by GodGardensGuns** [supabase/supabase#4565] [Questions]
I'd like to implement a search function that is able to look through multiple tables at once. I found [full text search](https://supabase.com/docs/guides/database/full-text-search) in the docs, but there's no example for querying multiple tables using a join or other method.

For example, I might want to search through `companies` and `products` for `baked beans in Boston`. I imagine the query would look something like this:

```js
const { data, error } = await supabase
  .from('companies')
  .select(`
    id,
    location,
    products:product_id ( title )
`)
  .textSearch(`'location' | 'title'`, `'baked beans' | 'Boston'`)
```

I haven't had a chance to test this, but I figure I may be pushing the limits of this feature. Is there a better way to go about doing this?

**Comment by steve-chavez** [comment]
Maybe try with:

```js
const { data, error } = await supabase
  .from('companies')
  .select(`
    id,
    location,
    products:product_id ( title )
`)
  .textSearch('location', `'baked beans' | 'Boston'`)
  .textSearch('products.title', `'baked beans' | 'Boston'`)
```

**Reply by KirioXX** [reply]
Wouldn't this mean that the keywords need to be in both columns to match?
Is there maybe a way to use `.filter()` to build a text search query over multiple columns?

**Reply by steve-chavez** [reply]
@KirioXX Yes, you can use `or` for that. See https://supabase.com/docs/reference/javascript/or (make sure to check the notes).

Should be like:

```
.or('location.fts.val, products.fts.another_val')
```

**Reply by KirioXX** [reply]
Awesome! Thanks, @steve-chavez. I'll give this a try.

**Reply by ajgil** [reply]
@KirioXX did you solved that or query?

**Reply by KirioXX** [reply]
@ajgil We decided against using the filter on select, instead we created search RPC functions. 
With that, we get more control over joins and filters. 
Plus RPC still supports modifiers like limit and order when it returns a table result.

**Reply by ajgil** [reply]
something like this post? [https://stackoverflow.com/questions/69225396/is-there-a-way-to-perform-full-text-search-on-multiple-columns-on-supabase-with](url)

My postgresql query runs perfect and i am thinking create materilized views or rpc function like your
my sql example
```sql
select e.title, e.description, m.place_name, m.text_name 
from eventos e
INNER JOIN tmp_maps AS m ON e.id = m.evento_id
where e.done = false
and
e.fts @@ to_tsquery('mobile | desktop')
or
m.fts @@ to_tsquery('madrid | brasil')
```

would be fine if you can text your rpc model

**Reply by KirioXX** [reply]
Yeah, that looks good. 

But I think you also need to use `to_tsvector`, like this:

```sql
select e.title, e.description, m.place_name, m.text_name 
from eventos e
INNER JOIN tmp_maps AS m ON e.id = m.evento_id
where e.done = false
and
to_tsvector(e.fts) @@ to_tsquery('mobile | desktop')
or
to_tsvector(m.fts) @@ to_tsquery('madrid | brasil')
```

Sorry, I sadly don't have a good example, most of our queries are quite complex or internal.
